runtime/shared_common/shrinit.cpp

```diff
@@ -5344,4 +5345,16 @@ findExistingCacheLayerNumbers(J9JavaVM* vm, const char* ctrlDirName, const char*
 	return;
 }
 
+/**
+ * Disables reads/writes to the shared class cache and marks the shared class cache as disabled during a CRIU restore
+ *
+ * @param [in] vm  The Java VM
+**/
+void
+j9shr_disableSharedClassCacheForCriuRestore(J9JavaVM* vm)
+{
+	vm->sharedClassConfig->runtimeFlags |= J9SHR_RUNTIMEFLAG_DENY_CACHE_ACCESS | J9SHR_RUNTIMEFLAG_DENY_CACHE_UPDATES;
```

Comment:
Can you check the APIs checking these 2 flags (`J9SHR_RUNTIMEFLAG_DENY_CACHE_ACCESS and J9SHR_RUNTIMEFLAG_DENY_CACHE_UPDATES`), we may need to update the return code. e.g.

https://github.com/eclipse-openj9/openj9/blob/03967d8cc24dc1f60168110c9d207d2dff57765a/runtime/shared_common/shrinit.cpp#L1820

The comment of this API says `Return -1 in the case of error`. But if `xShareClassCacheDisabledOnCRIURestore` is `TRUE`, it should not be an error case. It should behave as if the data is not in the shared cache and return 0.

Reply:
I just checked every usage of those two flags and I didn't find anything else that was incorrect, so I'll just fix that one.

Reply:
Fixed that function in [86ce34e](https://github.com/eclipse-openj9/openj9/commit/86ce34e9e82cb6489368e41b35512d4681859702)